### ISSUE TITLE
Implement `egctl`

### ISSUE THREAD

**Issue by zirain** [envoyproxy/gateway#915]
- [x] [design doc](https://github.com/envoyproxy/gateway/blob/main/docs/latest/design/egctl.md)
- [x] `version` command #914 
- [x] `config` command
  - [x] `config all` https://github.com/envoyproxy/gateway/pull/1060
  - [x] `config cluster` https://github.com/envoyproxy/gateway/pull/1083
  - [x] `config listener` https://github.com/envoyproxy/gateway/pull/1083
  - [ ] ?
- [ ] `analyze` command
- [ ] E2E tests
- [ ] Others?

**Comment by Xunzhuo** [comment]
Design: https://github.com/envoyproxy/gateway/blob/main/docs/latest/design/egctl.md

**Comment by Xunzhuo** [comment]
I see `egctl analyze` in desgin doc, what is the status of it?

**Comment by arkodg** [comment]
@zirain do you think we should address this in 0.4.0 or 0.5.0 (I prefer latter) ?

**Comment by zirain** [comment]
plan to support `version` and `config` in v0.4

**Comment by Xunzhuo** [comment]
How about adding support for:
+ `egctl install`
+ `egctl uninstall`
+ `egctl manifests`

After #651 gets merged?

**Comment by zirain** [comment]
IMO, we should consider add those once the install/uninstall became stable, support helm chart is enough for now.

BTW, setup e2e framework is more urgent.

**Comment by arkodg** [comment]
@zirain if all v0.4 items are tackled, can we move this issue to v0.5 ?

**Comment by arkodg** [comment]
closing in favor of https://github.com/envoyproxy/gateway/issues/2028
